src/controller/java/templates/ChipStructs-java.zapt

```diff
@@ -0,0 +1,58 @@
+{{> header}}
+{{#if (chip_has_client_clusters)}}
+package chip.devicecontroller;
+
+import androidx.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+
+public class ChipStructs {
+
+{{#zcl_clusters}}
+{{#zcl_structs}}
+{{#if itemCnt}}
+public static class {{asUpperCamelCase parent.name}}Cluster{{asUnderlyingType name}} {
+{{#zcl_struct_items}}
+public {{asJavaType type null parent.parent.name includeAnnotations=true}} {{asLowerCamelCase (asSymbol label)}};
+{{/zcl_struct_items}}
+
+  public {{asUpperCamelCase parent.name}}Cluster{{asUnderlyingType name}}(
+  {{#zcl_struct_items}}
+    {{#not_first}}, {{/not_first}}{{asJavaType type null parent.parent.name includeAnnotations=true}} {{asLowerCamelCase (asSymbol label)}}
+  {{/zcl_struct_items}}) {
+    {{#zcl_struct_items}}
+    this.{{asLowerCamelCase (asSymbol label)}} = {{asLowerCamelCase (asSymbol label)}};
+    {{/zcl_struct_items}}
+  }
+
+  @Override
+  public String toString() {
+    StringBuilder output = new StringBuilder();
+    output.append("{{asUpperCamelCase parent.name}}Cluster{{asUnderlyingType name}} {\n");
+    {{#zcl_struct_items}}
+    {{! TODO: Print more descriptive types for optional, nullable, array, and struct. }}
+    output.append("\t{{asLowerCamelCase name}}: ");
+    {{#if isList}}
+    output.append({{asLowerCamelCase name}});
```

Comment:
3 outputs below are the same.  Does handlebar have no or? It seems we want if isOctetString && !isList && !isArray.

What is the difference between a list and array? Does matter data model support both? I have only ever seen lists.

Reply:
Handlebars doesn't have an OR. We could implement our own helper in the future, but at least in this case it's not a big problem to not have an OR.

Not that clear on the difference myself. @bzbarsky-apple can you chime in here?

Reply:
> What is the difference between a list and array?

At one point, ZAP by default would flag lists as `isArray` _except_ for lists that are attributes, which it did not flag in any sort of sane way (just by having an `entryType`).  The chip-specific helpers then started setting `isList` for attributes.

Then we started setting `isArray` for attributes too, and fixing up the places that relied on `isList`....

I think at this point we could probably remove `isList` and switch all consumers to `isArray`.  I'll see about doing that.